docs/arcaflow/workflows/expressions.md

```diff
@@ -28,31 +85,274 @@ $.foo.bar
+
+Example:
+```JavaScript
+thisIsAFunction("this is a string literal for the first parameter", $.a.b)
 ```
-$.foo[0]
-```
 
-In Arcaflow, the two are equivalent.
+## Binary Operations
+
+Binary Operations have an expression to the left and right, with an operator in between.
+The order of operations determines which operators run first. See [Order of Operations](#order-of-operations)
+
+| Operator| Description        |
+| --------|--------------------|
+| `+`     | [Addition/Concatenation](#additionconcatenation) |
+| `-`     | [Subtraction](#subtraction) |
+| `*`     | [Multiplication](#multiplication)|
+| `/`     | [Division](#division)|
+| `%`     | [Modulus](#modulus)|
+| `^`     | [Exponentiation](#exponentiation)|
+| `==`    | [Equal To](#equal-to)|
+| `!=`    | [Not Equal To](#not-equal-to)|
+| `>`     | [Greater Than](#greater-than)|
+| `<`     | [Less Than](#less-than)|
+| `>=`    | [Greater Than or Equal To](#greater-than-or-equal-to)|
+| `<=`    | [Less Than or Equal To](#less-than-or-equal-to)|
+| `&&`    | [Logical And](#logical-and)|
+| `\|\|`  | [Logical Or](#logical-or)|
+
+
+### Addition/Concatenation
+
+This operator has different behavior depending on the type.
+
+##### String Concatenation
+
+When the `+` operator is used with two strings, it concatenates them together.
+For example, the expression `"a" + "b"` would output the string `"ab"`.
+
+##### Mathematical Addition
+
+When the `+` operator is used with numerical operands, it adds them together. The operator requires numerical operands with the same type. You cannot mix float and integer operands.
+For example, the expression `2 + 2` would output the integer `4`.
+
+
+### Subtraction
+
+When the `-` operator is used with a numerical operands, subtracts the value of the right operand from the value of the left. The operator requires numerical operands with the same type. You cannot mix float and integer operands.
+
+For example, the expression `6 - 4` would output the integer `2`.
+The expression `$.a - $.b` would evaluate the values of `a` and `b` within the root, and subtract the value of `$.b` from `$.a`.
+
+### Multiplication
+
+When the `*` operator is used with a numerical operands, it multiplies them. The operator requires numerical operands with the same type.
+
+For example, the expression `3 * 3` would output the integer `9`.
+
+### Division
+
+When the `/` operator is used with a numerical operands, divides the value of the left operand into the right operand. The operator requires numerical operands with the same type.
+
+The output type matches the input type. Integer division results in the value being rounded down into the resultant integer. If a non-whole number output is required, or if different rounding logic is required, convert the inputs into floating point number with TO BE ADDED BEFORE MERGE.
+
+For example, the expression `3 / 3` would output the integer `1`.
```

Comment:
`4 / 3 = 1` might be more useful here...

Reply:
Added a more useful example.